kclvm/sema/src/resolver/lint.rs

```diff
@@ -0,0 +1,608 @@
+//!    
+//!     ```rust,no_run
+//!    pub static IMPORT_Position: &Lint = &Lint {
+//!         ...
+//!    }
+//!    ```
+//!
+//! 2. Define a lintpass, which is used to implement the checking process，e.g.,
+//!    
+//!     ```rust,no_run
+//!    declare_lint_pass!(ImportPosition => [IMPORT_POSITION]);
+//!    ```
+//!
+//!    The `ImportPosition` is the defined LintPass structure and the `IMPORT_POSITION` is the `Lint` structure
+//!    defined in step 1. Here is a `LintArray`, which means that multiple lint checks can be implemented
+//!    in a single lintpass.
+//!
+//! 3. Implement the lintpass check process, e.g.,
+//!
+//!    ```rust,no_run
+//!    impl LintPass for ImportPosition{
+//!        fn check_module(&mut self, diags: &mut IndexSet<Diagnostic>, ctx: &mut LintContext,module: &ast::Module){
+//!            ...
+//!        }
+//!    }
+//!    ```
+//!
+//! 4. Add the `check_*` methods in lintpass to the macro `lint_methods`, or skip it if it exists
+//!
+//!    ```rust,no_run
+//!    macro_rules! lint_methods {
+//!        ($macro:path, $args:tt) => (
+//!            $macro!($args, [
+//!                fn check_module(module: &ast::Module);
+//!            ]);
+//!        )
+//!    }
+//!    ```
+//!
+//! 5. Add the new lintpass to the macro `default_lint_passes`, noting that `:` is preceded and followed by
+//! the name of the lintpass. e.g.,
+//!
+//!    ```rust,no_run
+//!    macro_rules! default_lint_passes {
+//!        ($macro:path, $args:tt) => {
+//!            $macro!(
+//!                $args,
+//!                [
+//!                    ImportPosition: ImportPosition,
+//!                ]
+//!            );
+//!        };
+//!    }
+//!    ```
+//!
+//! 6. If new `check_*` method was added in step 4, it needs to override the walk_* method in Linter.
+//! In addition to calling the self.pass.check_* function, the original walk method in MutSelfWalker
+//! should be copied here so that it can continue to traverse the child nodes.
+//!
+
+use std::collections::HashSet;
+
+use super::Resolver;
+use indexmap::{IndexMap, IndexSet};
+use kclvm_ast::ast;
+use kclvm_ast::walker::MutSelfWalker;
+use kclvm_ast::{walk_if, walk_list};
+use kclvm_error::{Diagnostic, DiagnosticId, Level, Message, Position, Style, WarningKind};
+use regex::Regex;
+
+/// A summary of the methods that need to be implemented in lintpass, to be added when constructing new lint
+/// lint and lintpass. When defining lintpass, the default implementation of these methods is provided: null
+/// check (see macro `expand_default_lint_pass_methods`). So what need to do is to override the specific
+/// `check_*` function.
+#[macro_export]
+macro_rules! lint_methods {
+    ($macro:path, $args:tt) => (
+        $macro!($args, [
+            fn check_module(module: &ast::Module);
+            fn check_module_post(module: &ast::Module);
+            /*
+            * Stmt
+            */
+
+            // fn check_stmt(stmt: ast::Node<ast::Stmt>);
```

Comment:
Why should this part of the code be commented out? If there is no error, you can leave them uncommented.

Reply:
Avoid useless empty functions, add them when needed